api/tools/src/main/java/org/pmiops/workbench/tools/cdrConfig/UpdateCdrConfig.java

```diff
@@ -0,0 +1,331 @@
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * See api/project.rb update-cdr-config. Reads Access Tiers and CDR versions from a JSON file and
+ * updates the database to match.
+ */
+@Configuration
+public class UpdateCdrConfig {
+
+  private static final Logger logger = Logger.getLogger(UpdateCdrConfig.class.getName());
+
+  @Bean
+  @Transactional
+  public CommandLineRunner run(
+      AccessTierDao accessTierDao, CdrVersionDao cdrVersionDao, WorkspaceDao workspaceDao)
+      throws IOException {
+    return (args) -> {
+      if (args.length != 2) {
+        throw new IllegalArgumentException(
+            "Expected 2 args (file, dry_run). Got " + Arrays.asList(args));
+      }
+
+      // it's small and we're parsing it multiple times, so read it into memory
+      final String configFile = new String(Files.readAllBytes(Paths.get(args[0])));
+      boolean dryRun = Boolean.parseBoolean(args[1]);
+
+      final Map<Long, Long> accessTierMap = generateAccessTierMap(configFile);
+
+      // no adapters necessary - Gson can infer the structure of DbAccessTier
+
+      final Gson accessTiersGson = new GsonBuilder().create();
+      final AccessTierExtractor accessTiers =
+          accessTiersGson.fromJson(configFile, AccessTierExtractor.class);
+
+      // DbCdrVersion requires Gson adapters for Timestamp formatting and DbAccessTier resolution
+
+      final Gson cdrVersionsGson =
+          new GsonBuilder()
+              .registerTypeAdapter(Timestamp.class, new TimestampGsonAdapter())
+              .registerTypeAdapter(DbAccessTier.class, new DbAccessTierIdGsonAdapter(accessTierDao))
+              .create();
+      final CdrVersionExtractor cdrVersions =
+          cdrVersionsGson.fromJson(configFile, CdrVersionExtractor.class);
+
+      preCheck(accessTiers.accessTiers, cdrVersions.cdrVersions, accessTierMap);
+
+      updateDB(
+          dryRun,
+          accessTiers,
+          accessTiersGson,
+          cdrVersions,
+          cdrVersionsGson,
+          accessTierMap,
+          accessTierDao,
+          cdrVersionDao);
+    };
+  }
+
+  // return a map of CDR Version IDs to Access Tier IDs, for later DB resolution
+  private Map<Long, Long> generateAccessTierMap(String configFile) {
+    Gson gson = new GsonBuilder().create();
+
+    final AccessTierMappingExtractor mapper =
+        gson.fromJson(configFile, AccessTierMappingExtractor.class);
+
+    Map<Long, Long> accessTierMap = Maps.newHashMap();
+    mapper.cdrVersions.forEach(v -> accessTierMap.put(v.cdrVersionId, v.accessTier));
+    return accessTierMap;
+  }
+
+  /**
+   * Check the file for internal consistency
+   *
+   * <p>Access Tiers cannot duplicate IDs
+   *
+   * <p>CDR Versions cannot:
+   *
+   * <p>- duplicate IDs
```

Comment:
nit: can you use `<li>` instead here? Presumably spotless broke your original formatting

Reply:
indeed, thanks